Jaden taking over for ShravyaKudlu Allow adding CC email with Enter key (DONE Jaden)

ShravyaKudlu opened the pull request:
Jaden taking over for ShravyaKudlu Allow adding CC email with Enter key (DONE Jaden)- #4002

## Description
Application/Job Posting Page/Function and Design: Allow adding CC email with Enter key. Fix Invalid Date in Date Posted column. Fix Manage CCs and Add Email CC to Category modal layout.

**Related #:** #2924, #1165, #3592

## Main changes explained
- Allow adding CC email via Enter key or Add Email button in both the Manage CCs and Add Email CC to Category modals
- Fix "Invalid Date" displaying in the Date Posted column when a job has no datePosted value
- Fix broken modal layout in JobCCModal and JobCategoryCCModal by replacing custom div wrappers with Reactstrap ModalHeader/ModalBody/ModalFooter

## How to test
1. Check out branch, run `npm install` and start the app locally
2. Clear site data/cache
3. Log in as Owner
4. Navigate to the Job CC Dashboard
5. Type a valid email in the Manage CCs modal and press Enter or click Add Email - email should appear immediately, input clears, no page refresh
6. Type a valid email in the Add Email CC to Category modal and press Enter or click Add Email - same behaviour
7. Verify Date Posted column shows a valid date or `-` (not "Invalid Date")
8. Verify both modals render correctly with a proper header, body, and footer in both light and dark mode

ShravyaKudlu committed:
Fixed the enter key when adding new email

ShravyaKudlu committed:
Fixed inline styling

ShravyaKudlu committed:
Fixed inline styling for jobCCModel.jsx and added

ShravyaKudlu committed:
Fixed inline styling for jobCCCatogory.jsx and added styling

sohamsharma08 reviewed (approved):
Tested the PR using an owner account. The add CC email feature is working as expected in both light and dark mode. Relevant screenshots are attached below:

<img width="1301" height="710" alt="PR 4002 (1)" src="https://github.com/user-attachments/assets/47dfbe6c-bac7-4af4-87c1-05b74a3e0906" />
<img width="1301" height="710" alt="PR 4002 (2)" src="https://github.com/user-attachments/assets/7ba6e031-7150-476c-a6d2-9f2f5b374aeb" />
<img width="1301" height="710" alt="PR 4002 (3)" src="https://github.com/user-attachments/assets/8135cf8a-4640-40e1-8e19-a204a33d2619" />
<img width="1301" height="710" alt="PR 4002 (4)" src="https://github.com/user-attachments/assets/77d7791a-3ac3-4236-808f-8fb70783000e" />
<img width="1301" height="710" alt="PR 4002 (5)" src="https://github.com/user-attachments/assets/47f411c7-46c1-44ac-9dd4-9cc8ca4c0ba7" />

Aditya-gam reviewed (changes requested):
- The add email to CC category correctly opens the modal and adds the email to the correct category list.
- The Manage CC button also correctly opens the modal with the current list and lets me add a new email.
- The Page looks visually correct for both light and dark mode.

**These are the issues I found**
- When adding an email using Add Email to CC Category, the email is successfully added, but there is no toast or notification of success or failure; instead, the email field is reset to empty, and the category field still holds that selected category.
- I cannot add the email using the `Enter Key` for the `Add Email CC to Category` button, but it works for the `Manage CC` button.
- If I add an email to a specific job title using the Manage CC button and then immediately try deleting it without closing the modal, it throws an error. It doesn't let me delete the email entry. I can delete the same entry once I close the modal and reopen it.
- As the screens get smaller(eg, iPad/phone), the content of the page gets clipped. This should be made responsive.

https://github.com/user-attachments/assets/4b632557-44cc-45d9-84d8-09b601c3e32e

Nhujarski reviewed (changes requested):
Tested the PR using an owner account. The add CC email feature is working as expected in both light and dark mode, however I did notice when I press the enter key the email is added, but there is a refresh of the page. I couldn't get a screenshot of it. Everything else is looking good and functioning correctly. 

<img width="1440" height="777" alt="Job cc light" src="https://github.com/user-attachments/assets/c9748e09-9a89-43b6-811a-745c496f0be1" />
<img width="1437" height="882" alt="JOb cc dark" src="https://github.com/user-attachments/assets/c1efca8d-47a8-4a42-9d7b-097184900110" />
<img width="1421" height="754" alt="Job cc modal" src="https://github.com/user-attachments/assets/3e39d1c4-e0d4-45c3-9fc1-593cc5120e56" />
<img width="1440" height="765" alt="Job cc email added" src="https://github.com/user-attachments/assets/f12b25ae-e81c-4c04-882e-03a6b40303cd" />

harshavarma29 reviewed (dismissed):
I have tested the PR locally using the owner account, and it is working as expected. Both the "Add Email" button and the Enter key are functioning correctly.

1) Adding an email using the "Add Email" button:

<img width="637" height="496" alt="add_email_working_fine" src="https://github.com/user-attachments/assets/7da8e08c-5844-4861-b90e-c94e45f70210" />

2) Adding an email using the Enter key:

<img width="636" height="492" alt="enter_key_working_fine" src="https://github.com/user-attachments/assets/2c5f0a2f-ea3e-4266-ac00-a2e7a74d8f42" />

3) Details added successfully:

<img width="1857" height="666" alt="working_fine" src="https://github.com/user-attachments/assets/706be6f8-7335-455c-a7bd-d5ca5028f03a" />

laynet reviewed (dismissed):
I followed testing instructions and was able to verify that when I added an email to a category that the email appears in the list immediately, the input clears, there was no page refresh and I can successfully use both the submit button and the enter button. I also verified this in dark mode.
![PR 4002 ui verified](https://github.com/user-attachments/assets/c1b5130a-70e0-4920-b94e-593cfe1917f9)

IndraAnuraag9 commented:
I followed the testing instructions and tested this PR using an Owner Account. 
I was able to add an email by using both the "Add Email" Button and using the Enter Key. This works both in Light and Dark Mode. 

Issues I noticed:
1. When adding an email using the "Add Email CC to Category", there is no confirmation for success/failure, and the email field is reset to empty, and the category field still has the selected category.
2. I am able to add an email using the "Manage CC". But if I try to delete it immediately, it throws an error. I have to close the modal, re-open it to delete it.

<img width="1440" height="900" alt="Screenshot 2025-10-02 at 6 31 08 PM" src="https://github.com/user-attachments/assets/a72803a6-2394-48b1-918c-72633cb0439f" />
<img width="1440" height="900" alt="Screenshot 2025-10-02 at 6 31 31 PM" src="https://github.com/user-attachments/assets/6ff24adf-4aa1-419f-88ed-fc0da8cbb149" />
<img width="1440" height="900" alt="Screenshot 2025-10-02 at 6 32 14 PM" src="https://github.com/user-attachments/assets/b654285d-d95b-4113-9d05-3550393d1218" />
<img width="1440" height="900" alt="Screenshot 2025-10-02 at 6 32 44 PM" src="https://github.com/user-attachments/assets/11035c34-d62c-4e5e-8427-d65efc72d30f" />
<img width="1440" height="900" alt="Screenshot 2025-10-02 at 6 32 59 PM" src="https://github.com/user-attachments/assets/97f9b302-bc69-493d-b8be-70550065ab68" />
<img width="1440" height="900" alt="Screenshot 2025-10-02 at 6 33 07 PM" src="https://github.com/user-attachments/assets/4abbdfe6-5446-4231-b4be-f67ae2b0ad8f" />

https://github.com/user-attachments/assets/2a54a839-2b25-4ac7-873f-d17e0817101a

Anusha-Gali reviewed (changes requested):
Hi Shravya,

I have reviewed your PR locally and though the main functionality works perfectly i.e able to add email via button and via enter as well as works perfectly in dark mode. Though I have identified two issues.
1) The add email does not verify the email added based on any regex
2) If we try to remove the just added email. It does not let us do so. We need to close and open Manage CC's again to remove an email
- Also would like to understand the date posted concept, should it get updated when a new CC is added.
<img width="1276" height="741" alt="Screenshot 2025-11-21 at 8 36 48 PM" src="https://github.com/user-attachments/assets/a9c40a59-fd7a-4ba3-b261-11deabc9f7a2" />
<img width="1276" height="741" alt="Screenshot 2025-11-21 at 8 37 22 PM" src="https://github.com/user-attachments/assets/fcd38b56-73a8-4406-b146-23667b172e03" />
<img width="1276" height="174" alt="Screenshot 2025-11-21 at 8 38 13 PM" src="https://github.com/user-attachments/assets/565a8589-f74f-40b8-a8cb-d3f150cf37a3" />
<img width="1276" height="414" alt="Screenshot 2025-11-21 at 8 38 41 PM" src="https://github.com/user-attachments/assets/915e9030-fb5b-41de-8017-6875a0c6ec47" />
<img width="1276" height="414" alt="Screenshot 2025-11-21 at 8 38 50 PM" src="https://github.com/user-attachments/assets/ac174bb3-a030-4a07-a9d9-fb0a6f0e68c9" />

rohanrastogi311 reviewed (approved):
Hi Shravya, 

Well done with this implementation. 

<img width="1728" height="1046" alt="Image" src="https://github.com/user-attachments/assets/ff41b7e9-88f1-4dcd-94d5-5bce0a5f31ef" />
<img width="1728" height="1047" alt="Image" src="https://github.com/user-attachments/assets/d1a36e3b-108a-49e3-bc9a-832fec20883e" />

ShravyaKudlu committed:
Merge branch 'development' into ShravyaKudlu_Feature/Allow_adding_CC_email_with_Enter_key

ShravyaKudlu commented:
@Anusha-Gali, Please ensure to test according to the description provided, Thanks

DeepighaJ reviewed (changes requested):
Checked out to current branch and getting error Code: MODULE_NOT_FOUND ON npm run start:local

<img width="807" height="552" alt="Image" src="https://github.com/user-attachments/assets/4faf4e57-c1d9-4652-9bb1-c214942be616" />

HemanthNidamanuru reviewed (changes requested):
Hi Shravya,

I am getting a "Something went wrong" error when navigating to the page. Could you please check?

<img width="1919" height="1090" alt="Image" src="https://github.com/user-attachments/assets/43b69066-7f92-4b2f-b0d0-7b4099524298" />

ShravyaKudlu committed:
fix: merge development

ShravyaKudlu committed:
fix: comments

ShravyaKudlu committed:
fix: sonarqube props

ShravyaKudlu committed:
fix: sonarqube props again

ShravyaKudlu committed:
fix: sonarqube color contrast

kzou55 reviewed (changes requested):
Hi ShravyaKudlu, 

I ran and tested the PR locally. 

## Verified
- Email form pops up either through the "Add Email CC to Category" or "Manage CCs" button
- Can add email using the "Add Email" button or by pressing "Enter"
- Email appears in the list immediately, input clears, no page refresh occurs
- Dark Mode works and looks fine

- Tests
  - Testing in Light Mode

    https://github.com/user-attachments/assets/a3fb7fa9-a00a-4d1a-8846-6bdfbec4bb3f

  - Testing in Dark Mode

    https://github.com/user-attachments/assets/68d6baab-6df9-4831-afe2-4b14c3a42ebc

## Issues
1. When inputting an invalid email, the pressing "Enter" method checks if the email is validated. This fails when an invalid email is added when using the click "Add Email" button. Not sure if this is within scope of the PR. 

    https://github.com/user-attachments/assets/560f69d9-dd22-4d5e-a2c0-7ef5098f555c

DeepighaJ reviewed (changes requested):
Tested the PR locally. I'm able to add Email as given mentioned in the steps but i see date as Invalid Date in the added list. 

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/fc9c41c1-fbe0-47fd-9a45-662386feae23" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/89c07f02-d7b2-4140-a51b-6ac991bc6647" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/8c573342-6b6b-4a2b-a626-3861c276e71c" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/fb61528e-291a-4fc0-977c-ec498979c7ee" />
<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/178226e6-2327-414a-9452-94fc807c38fc" />

HemanthNidamanuru reviewed (changes requested):
Hi Shravya,

I tested this PR locally.

The new feature is working as expected. I was able to add a valid email using Enter and the Add Email button. The email was added immediately, the input cleared, and there was no page refresh.

Dark mode also looks fine.

One thing I noticed is that the date is showing as “Invalid Date” for some entries.

<img width="1918" height="1035" alt="Image" src="https://github.com/user-attachments/assets/3244f212-c322-4b82-8ea0-085cbd03d8b8" />

<img width="1897" height="851" alt="Image" src="https://github.com/user-attachments/assets/66983eb0-aafb-4cbd-b707-73911cbc11f9" />

ShravyaKudlu committed:
fix: validate email

ShravyaKudlu committed:
fix: sonarqube simplified

kzou55 reviewed (changes requested):
Hi Shravya, 

I ran and retested the PR locally

## Verified
- Email can be added through the Enter key in both "Add Email CC to Category" or "Manage CCs"
- Email updated instantly and without refresh
- Works in Dark Mode
- Email Error check
  - Through "Add Email CC to Category" 
    <img width="1252" height="425" alt="Image" src="https://github.com/user-attachments/assets/b6656a58-9e1a-426a-85ff-6a6d48ab9f8e" />
  - Through Manage CCs
    <img width="1303" height="657" alt="Image" src="https://github.com/user-attachments/assets/24396c22-bc92-41b0-a0c1-8e831a48eb38" />  


## Issues 
- Previous review this was fine, but now the Manage CCs pop up has some layout issues. 

  <img width="645" height="686" alt="Image" src="https://github.com/user-attachments/assets/1f65e7ce-ba40-4f90-b768-630f4a6f66c4" />

## Note
- The "Invalid Date" for the Date Posted column is still present, assuming this will be a separate bug issue and handled later/elsewhere

HemanthNidamanuru reviewed (approved):
Hi Shravya,

Thanks for the clarification.

I rechecked the PR based on the required scope. The add CC email feature is working as expected. I was able to add a valid email using both Enter and the Add Email button, and the email appears immediately without page refresh.

Dark mode also looks fine.


<video src="https://github.com/user-attachments/assets/ceb26ded-e86c-4d12-825f-3e90be861e1d" controls width="700"></video>


<img width="1917" height="1030" alt="Image" src="https://github.com/user-attachments/assets/b18b9025-6d29-4175-a780-5c0c5644b769" />

<img width="1917" height="1042" alt="Image" src="https://github.com/user-attachments/assets/04e8b0d9-5fbb-4fe8-ae5a-03f77750e5e3" />

Jaden300 committed:
fix(JobCCDashboard): fix Invalid Date display and modal layout regression

- Guard datePosted with fallback to '-' to prevent Invalid Date rendering
- Replace custom div-based modal structure with Reactstrap ModalHeader/ModalBody/ModalFooter in JobCCModal and JobCategoryCCModal

Jaden300 committed:
Fix inline style, delete-after-add bug, and remove stale comments in JobCCDashboard

DeepighaJ reviewed (approved):
Tested the PR locally.
The functionality works as expected.
The invalid date issue is fixed and looks good.

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/92b5a22b-a120-482f-bea3-483af199baa7" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/bcd0e5d7-88f3-43ef-be3c-fe3be2f8bf79" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/fe4fa4b5-d70e-4060-8ebb-f3e9dddf5252" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/31331332-4322-4d46-a570-6ed6d6bd381f" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/3290784a-cc8f-4f20-9170-723fc1ef0ae2" />

<img width="1600" height="852" alt="Image" src="https://github.com/user-attachments/assets/d1b5212e-d44b-480d-a336-941c708ad479" />

kzou55 reviewed (approved):
Hi Jaden, 

I ran and tested the PR locally. 

## Verified
- Can type a valid email in the Manage CCs modal and press Enter or click Add Email - email should appear immediately, input clears, no page refresh
- Can type a valid email in the Add Email CC to Category modal and press Enter or click Add Email - email should appear immediately, input clears, no page refresh
-  Date Posted column shows a valid date or - (not "Invalid Date")
- Both modals render correctly with a proper header, body, and footer in both light and dark mode

## Testing
- Light Mode

  https://github.com/user-attachments/assets/b7c03367-cb44-4515-b3b9-b55fe0dd802a

- Dark Mode

  https://github.com/user-attachments/assets/f895172c-accb-45fa-ad2a-040fbba4f88e
